### ISSUE TITLE
suite.T() no longer compatible with *testing.T (compatibility changed)

### ISSUE THREAD

**Issue by narqo** [stretchr/testify#893]
Probably, following #867

In 1.5.0 the behaviour of `Suite.T()` has been changed in a backwards-incompatible way, even though the release notes say there are only non-breaking changes.

Before 1.5.0 `Suite.T()` returned `*testing.T` while it returns custom `TestingT` interface now. So the code below is broken (_note, the code is massively simplified from the real use cases_):

```go
type FooSuite struct {
    suite.Suite
}

func (suite *FooSuite) TestSomething() {
    // Cannot use 'suite.T()' (type TestingT) as type *testing.T 
    testSomethingT(suite.T())

    // Cannot use 'suite.T()' (type TestingT) as type testing.TB Type does not implement 'testing.TB'   
    // as some methods are missing: Error(args ...interface{}) Fail() Failed() bool ...
    testSomethingTB(suite.T())
}

func testSomethingT(t *testing.T) {}

func testSomethingTB(t testing.TB) {}
```

**Comment by narqo** [comment]
A small note, the master version of the documentation no longer reflects the behaviour https://godoc.org/github.com/stretchr/testify/suite#Suite.T:

> T retrieves the current *testing.T context.

**Comment by glesica** [comment]
If you submit a PR we can do a 1.5.1 release.

**Comment by narqo** [comment]
I'd be happy, but I don't understand the direction in which suite's API goes. Shall I try to rollback the switch to interfaces in suite or is `suite.testingT` was expected to be `testing.TB`-compatible?

**Comment by glesica** [comment]
I know there was some discussion around this, but I don't recall the outcome, @boyan-soubachov do you remember? If I get a minute today I'll try to find the PR I'm thinking of.

**Comment by boyan-soubachov** [comment]
We should definitely revert #867 then. Excuse me, my mistake. Didn't realise that we return `*testing.T` intentionally there

**Comment by boyan-soubachov** [comment]
#894 created to revert this. Thorough reviews and feedback would be greatly appreciated 🙏

**Comment by narqo** [comment]
I believe this one was fixed with 1.5.1
